### DISCUSSION TITLE
Ussing easyeffects with a retrogaming distribution

### DISCUSSION THREAD

**Discussion by albmm74** [wwmm/easyeffects#2192] [General]
Hi friend.

I'd like to use easyeffects to normalize the sound output in a retrogaming distribution based on Ubuntu 22.04, to get most emulators/games sound at more o less the same volume.

Could this be possible? Does it work in realtime?

Thanks in advice.

**Comment by wwmm** [comment]
>to get most emulators/games sound at more o less the same volume.

For my personal needs the `autogain` plugin in EasyEffects is enough. But you may also want to take a look at presets like https://github.com/Digitalone1/EasyEffects-Presets/blob/master/LoudnessEqualizer.json that are more aggressively changing the audio dynamic range.

>Does it work in realtime?

The plugins work in a realtime thread configured by PipeWire.  If the system will be able to keep up it will depend on how powerful it is.

**Comment by albmm74** [comment]
Many thanks.

Considering I need only this feature, what are the minimal plugins/dependencies I need to install?

Thanks again.

**Reply by wwmm** [reply]
>what are the minimal plugins/dependencies I need to install?

I do not think there is room to save a lot of space. The autogain is a built-in plugin that is complied with EasyEffects. Not way to disable it for now. Most of the other plugins are from `Calf` or `Linux Studio Plugins`. The preset in the link uses plugins from `Linux Studio Plugins`. So you need it if you want to try that preset.

**Comment by albmm74** [comment]
Thanks again friend, but I can compile it on Ubuntu 22.04, getting a message saying libportal-gtk4 is not present; tried to install libgtk-4-dev broke my distribution...

I found a version of pulseeffects in the oficials ubuntu repositories, could be valid for my propossal?

UPDATE: working good with your recomended digitalone preset. Many thanks!

**Reply by wwmm** [reply]
>Thanks again friend, but I can compile it on Ubuntu 22.04, getting a message saying libportal-gtk4 is not present; tried to install libgtk-4-dev broke my distribution...

One solution to consider when the distribution does not provide the packages for a native EasyEffects build is using the Flatpak package https://github.com/wwmm/easyeffects/wiki/Package-Repositories#flatpak.

**Comment by albmm74** [comment]
Thanks Wellington.

Really testing it slowly it is not working as I want, I think due the digitalone profile do not use autogain, and some old arcade (MAME) games do not raise its volumen levels, what is what I want.

Appart of this, launching pulseeffects on a terminal I get the rnnoise plugin error, and I do not get it working with the -w option to hide the windows (and really I need it be hide when I launch it from .xinitrc file).

Can you help me?

Thanks in advice.

**Reply by wwmm** [reply]
>and really I need it be hide when I launch it from .xinitrc file

In this case it is probably better to start it with the command line option `--gapplication-service` so that it doesn't even show its window and start in service mode.

**Comment by albmm74** [comment]
I got this message from terminal launching pulseeffects --gapplication-service :

Failed to register: Unable to acquire bus name 'com.github.wwmm.pulseeffects'

**Comment by lonewolf21484-cyber** [comment]
the newest update has me ready to break something.  the mouse wheel doesn't work right, barely moves the sliders.  looking for alternative to easy effects now
